Updates to example PSE for C26 (#68)
Ada-Developers-Academy/core-pses: about-pses/example-pse.md

Reviewer: I know that this section hasn't been changed in this PR, but how do you feel about providing the link to the gist and then sharing the note about submitting a shareable link?

For me, the flow felt choppy.

```suggestion
Chat Link: [https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1](https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1)

*Note:*
*We are using a gist link in this example to ensure that the link exists from cohort to cohort. When submitting your response, the link should be a shareable link to your chat in the AI tool where you held the conversation.*
```

- String Sanitization: You may assume strings will not include invalid characters.

##### !end-question
##### !placeholder

> Add observations, assuptions, and questions here:

Reviewer: ```suggestion
Add observations, assumptions, and questions here:
```

1. From the problem description, I observe that...
2. From the example inputs I can see...
3. A question I have around inputs is...
4. ...

> 5. ...

Reviewer: These placeholders look good to me 👍 

Some other potential sentence starters could be:
- "Can I assume that...?"
- "What behavior is expected when...?"

> 5. ...

Author: ```suggestion
1. From the problem description, I observe that...
2. In the example inputs I can see...
3. A question I have around inputs is...
4. Based on the requirements, I am making the assumption...
5. What behavior is expected when...
```

    * What is the expected output for the given input?
    * You can use the examples provided in the prompt, or other examples.
2. Write unit tests for `winner` for the nominal and edge cases you identified in the first step.
    

> When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs. 

Reviewer: 👍 How would you feel about adding some examples to this section? Maybe like:

```suggestion
When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs. 

For example, if we wanted to test that the function `winner` returns a tie when `player_1` and `player_2` have the same value, then we might name the test something like `test_winner_returns_tie_for_players_with_same_value`.
```

I feel ok leaving this off too because we also have lessons about testing.

Author: I feel like including an example here would be a nice reminder and helps guide folks. Would you want to include the example in the question prompt on later PSEs, or just here on the example PSE?

Reviewer: If we include the example in subsequent question prompts, would they each have to be changed to relevant to the question being asked?

If so, then I think just adding the example in this file is ok.

Author: I don't think we need to update the example for each PSE, but we'd likely want to adjust the language to something like:

* e.g. from the example PSE: if we wanted to test that the function `winner` returns a tie when `player_1` and `player_2` have the same value, then we might name the test something like `test_winner_both_inputs_paper_returns_tie`.

<br>

Example steps:
1. First, we'll handle our simplest case which is checking for a tie: 

>     - if `player_1` is the same as `player_2`, return "It's a tie!"

Reviewer: I like how it visually looks to have the variables surrounded in tick marks so they appear differently in this long block of text. 

Could we also add tick marks around `player_1` and `player_2` on lines 288-296?

2. If it is not a tie, we'll move through the cases where `player_1` is "rock", "paper", or "scissors" to determine a winner, starting with the case that player_1 is "rock":
    - if player_2 is "scissors", return "Player 1 wins!"
    - otherwise player_2 must be "paper", return "Player 2 wins!"
3. Next we'll deal with the case that player_1 is "paper":
    - if player_2 is "rock", return "Player 1 wins!"
    - otherwise player_2 must be "scissors", return "Player 2 wins!"
4. Otherwise player_1 must be "scissors":
    - if player_2 is "rock", return "Player 2 wins!"

>     - otherwise player_2 must be "paper", return "Player 1 wins!" 

Author: ```suggestion
2. If it is not a tie, we'll move through the cases where `player_1` is "rock", "paper", or "scissors" to determine a winner, starting with the case that `player_1` is "rock":
- if `player_2` is "scissors", return "Player 1 wins!"
- otherwise `player_2` must be "paper", return "Player 2 wins!"
3. Next we'll deal with the case that `player_1` is "paper":
- if `player_2` is "rock", return "Player 1 wins!"
- otherwise `player_2` must be "scissors", return "Player 2 wins!"
4. Otherwise `player_1` must be "scissors":
- if `player_2` is "rock", return "Player 2 wins!"
    - otherwise `player_2` must be "paper", return "Player 1 wins!" 
```

Author: For some reason GitHub won't allow me to apply this suggestion, so I'll push the change up manually shortly.

> 
> <br>
> 
> One strength the AI highlighted was my use of "otherwise... must be" phrasing throughout my steps. The feedback noted that this showed I understood "that earlier conditions have already eliminated other possibilities," which is the kind of reasoning interviewers want to see. I think this worked well because it demonstrates I wasn't just listing cases mechanically, I was actively narrowing down what remained, which makes the logic easier to follow and verify.

Author: ```suggestion
Depending on exactly what the AI shares, a reflection and summary might look like:

<br>

Chat Link: [https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1](https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1)

<br>

*Note:*

*We are using a gist link in this example to ensure that the link exists from cohort to cohort. When submitting your response, the link should be a shareable link to your chat in the AI tool where you held the conversation.*

<br>

One strength the AI highlighted was my use of "otherwise... must be" phrasing throughout my steps. The feedback noted that this showed I understood "that earlier conditions have already eliminated other possibilities," which is the kind of reasoning interviewers want to see. I think this worked well because it demonstrates I wasn't just listing cases mechanically, I was actively narrowing down what remained, which makes the logic easier to follow and verify.
```
